middleware/proxy_test.go

```diff
@@ -393,6 +393,189 @@ func TestProxyError(t *testing.T) {
 	assert.Equal(t, http.StatusBadGateway, rec.Code)
 }
 
+func TestProxyRetries(t *testing.T) {
+
+	newServer := func(res int) (*url.URL, *httptest.Server) {
+		server := httptest.NewServer(
+			http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(res)
+			}),
+		)
+		targetUrl, _ := url.Parse(server.URL)
+		return targetUrl, server
+	}
+
+	targetUrl, server := newServer(http.StatusOK)
+	defer server.Close()
+	goodTarget := &ProxyTarget{
+		Name: "Good",
+		URL:  targetUrl,
+	}
+
+	targetUrl, server = newServer(http.StatusBadRequest)
+	defer server.Close()
+	goodTargetWith40X := &ProxyTarget{
+		Name: "Good with 40X",
+		URL:  targetUrl,
+	}
+
+	targetUrl, _ = url.Parse("http://127.0.0.1:27121")
+	badTarget := &ProxyTarget{
+		Name: "Bad",
+		URL:  targetUrl,
+	}
+
+	var alwaysRetryFilter ProxyRetryFilter = func(c echo.Context) bool { return true }
+	var neverRetryFilter ProxyRetryFilter = func(c echo.Context) bool { return false }
+
+	testCases := []struct {
+		name             string
+		retryCount       int
+		retryFilters     []ProxyRetryFilter
+		targets          []*ProxyTarget
+		expectedResponse int
+	}{
+		{
+			"retry count 0 does not attempt retry on fail",
```

Comment:
Could you please use field names. This way these structs are more future proof - when new fields are added the code would still compile, without names we would get compile errors.